Framework/script/RepoCleaner/qcrepocleaner/rules/1_per_run.py

```diff
@@ -68,16 +68,17 @@ def process(ccdb: Ccdb, object_path: str, delay: int,  from_timestamp: int, to_t
     # Dispatch the versions to deletion and preservation lists
     for bucket, run_versions in versions_buckets_dict.items():
         # logger.debug(f"- bucket {bucket}")
-        sorted_run_versions = sorted(run_versions, key=lambda x: (x.validFrom, -x.createdAt))
+        sorted_run_versions = sorted(run_versions, key=lambda x: (x.createdAt))
 
         freshest: ObjectVersion = None
         for v in sorted_run_versions:
-            if freshest is None or freshest.validFromAsDt < v.validFromAsDt:
-                if freshest is not None:
-                    if policies_utils.in_grace_period(freshest, delay):
-                        preservation_list.append(freshest)
-                    else:
-                        deletion_list.append(freshest)
+            if freshest is None:
+                freshest = v
```

Comment:
you are changing logic of the original loop. `freshest` was assigned only when it was valid, now you assign it every time when it is not `None`... so basically do you initialize it as the `sorted_run_versions[0]`? Is it correct? and if so, shouldn't it be done outside of the loop?

Reply:
oh no, sorry it is the same in the original loop, but weirdly written. So the `freshest` is the first item in `sorted_run_versions ` dataset everytime?

Reply:
basically yes. I rephrased it, thinking it's clearer to write it this way

Reply:
i just find it confusing to check if a variable is `None` in an `if` statement and then check if it is **not** `None` in another `if` statement inside.